pkg/api/types.go

```diff
@@ -1481,17 +1489,13 @@ const (
 	// onto the node entirely. Enforced by the scheduler.
 	TaintEffectPreferNoSchedule TaintEffect = "PreferNoSchedule"
 	// NOT YET IMPLEMENTED. TODO: Uncomment field once it is implemented.
-	// Do not allow new pods to schedule onto the node unless they tolerate the taint,
-	// do not allow pods to start on Kubelet unless they tolerate the taint,
+	// Do not allow pods to start on Kubelet unless they tolerate the taint,
 	// but allow all already-running pods to continue running.
-	// Enforced by the scheduler and Kubelet.
-	// TaintEffectNoScheduleNoAdmit TaintEffect = "NoScheduleNoAdmit"
-	// NOT YET IMPLEMENTED. TODO: Uncomment field once it is implemented.
-	// Do not allow new pods to schedule onto the node unless they tolerate the taint,
-	// do not allow pods to start on Kubelet unless they tolerate the taint,
-	// and evict any already-running pods that do not tolerate the taint.
-	// Enforced by the scheduler and Kubelet.
-	// TaintEffectNoScheduleNoAdmitNoExecute = "NoScheduleNoAdmitNoExecute"
+	// Enforced by Kubelet.
+	// TaintEffectNoAdmit TaintEffect = "NoAdmit"
+	// Evict any already-running pods that do not tolerate the taint.
+	// Enforced by Kubelet.
```

Comment:
We haven't decided anything. Depends on how autonomous we want Kubelet to be. Realistically I doubt we will have time to change the implementation any time in the foreseeable future, even if we wanted to. I agree the "unreachable" case has to go in the master, but the other ones theoretically don't. Anyway, the new comment I suggested does not imply we will change it; it just describes the current implementatino.